src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -45,20 +47,21 @@
  */
 final class DefaultRefreshable<T> implements SettableRefreshable<T> {
     private static final Logger log = LoggerFactory.getLogger(DefaultRefreshable.class);
-    /**
-     * Every ten times refreshable.map is called without an update we must purge SelfRemovingMapSubscriber instances
-     * whose children have been reaped.
-     */
-    private static final int CLEAN_THRESHOLD = 10;
+    private static final Cleaner REFRESHABLE_CLEANER = Cleaner.create();
 
     private static final int WARN_THRESHOLD = 1000;
 
-    @GuardedBy("this")
-    private final Set<Consumer<? super T>> orderedSubscribers = new LinkedHashSet<>();
+    // Subscribers are updated in deterministic order based on registration order. This prevents a class
```

Comment:
We have a fairly concise test here, I'm not sure of the best way to articulate that here. I prefer not to duplicate it, because it may need to change, and the test provides some level of enforcement. Similarly, referencing tests by name/link can grow stale over time as well. What do you think?
https://github.com/palantir/refreshable/blob/47170919fa247d57c285349e3a931dc56a6aef99/src/test/java/com/palantir/refreshable/RefreshableTest.java#L205-L212

Reply:
This has been updated great!